public/js/gf-giftaid-field-frontend.js

```diff
@@ -1,57 +1,93 @@
-function gfGiftAidOnInputChange(elem) {
-  if (!(elem instanceof HTMLElement)) {
-    return;
+/**
+ * The donation total object.
+ * @typedef {Object} Donation
+ * @property {string} donation - The amount donated as a string.
+ * @property {string} giftAidTotal - The amount including taxback as a string.
+ */
```

Comment:
Honestly, I think the code would be much more simple if you didn't use this!

Reply:
When I say this, I did not mean replace with untyped object.
My apologies.
I meant that instead of passing an object with properties, you should pass the donation amount and giftaid total as 2 separate arguments.
This way, you do not need to check object AND properties; only the values.